lib/swaps/Swaps.ts

```diff
@@ -96,6 +50,14 @@ class Swaps extends EventEmitter {
     return { takerAmount, makerAmount };
   }
 
+  public init = async () => {
+    // Load Swaps from data base
+    const result = await this.repository.getSwapDeals();
+    result.forEach((deal: SwapDeal) => {
```

Comment:
I think the simplest fix here is just to have this be `deal: SwapDealInstance`?

Reply:
I think we should have 2 types of `SwapDeal`s one for saving in memory and one for saving in the `DB`, currently Im getting errors here for example [error](https://github.com/ExchangeUnion/xud/blob/a8f7812b27fe466a5b365c2b40a5fb11a988fada/lib/swaps/Swaps.ts#L291).
The errors are because [of](https://github.com/ExchangeUnion/xud/blob/a8f7812b27fe466a5b365c2b40a5fb11a988fada/lib/types/orders.ts#L116) the use of `<type> | undefined`, for some reason [`attributes`](https://github.com/ExchangeUnion/xud/blob/a8f7812b27fe466a5b365c2b40a5fb11a988fada/lib/db/models/SwapDeal.ts#L36) will throw an error if I use for example `makerCltvDelta?: number;`

Reply:
`SwapDealInstance` *is* the type for what gets saved to the database. I haven't tried pulling this branch down yet, but would using that type for this line fix the issues?

Reply:
Yeah if I use 2 separate types it solves the issue.

Reply:
Ok cool, change this line when you get a chance and I'll review again.